Src/VsVim/VsVimPackage.cs

```diff
@@ -38,11 +41,21 @@ public VsVimPackage()
         {
         }
 
-        protected override void Initialize()
+        protected override async Task InitializeAsync(CancellationToken cancellationToken, IProgress<ServiceProgressData> progress)
         {
-            base.Initialize();
-
-            _componentModel = (IComponentModel)GetService(typeof(SComponentModel));
+            object componentModel = await GetServiceAsync(typeof(SComponentModel));
+            await JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+
+            // The cast to IComponentModel MUST occur after the call to SwitchToMainThreadAsync. The 
+            // call to GetServiceAsync can occur on any thread but the cast to the actual service 
+            // implementation must occur on the UI thread (for STA services at least). The implementation
+            // can go through QueryService if the object is a CCW and that must be done on the appropriate
+            // thread.
+            //
+            // For the services we use here they are likely to be managed objects and hence not really 
+            // applicable to this rule. But this is the best practice and should be followed in either
+            // case.
+            _componentModel = (IComponentModel)componentModel;

```

Comment:
It seems incredible that it is "this easy"!

Reply:
The hard part was figuring out this was all the code I actually needed here. 😦

Reply:
That's what I call brain surgery!